### PR TITLE
remove children autofocus & add Text Input focus

### PR DESCRIPTION
**Pull request by PierreCrb**
Related to #43

### CODE REVIEW

File: src/hooks/useInteractive.ts [hunk 1]
```diff
@@ -32,6 +31,12 @@ export const useInteractive = (
       event.preventDefault()
       event.stopPropagation()
       dispatch.components.select(component.id)
+      dispatch.app.toggleInputText()
```
Comment (tlenclos) [inline src/hooks/useInteractive.ts]: We should remove this

File: src/components/inspector/controls/ChildrenControl.tsx [hunk 2]
```diff
@@ -1,26 +1,45 @@
...
 
 const ChildrenControl: React.FC = () => {
-  const { setValueFromEvent } = useForm();
-  const children = usePropsSelector("children");
+  const dispatch = useDispatch()
+  const textInput = useRef<HTMLInputElement>(null)
+  const focusInput = useSelector(getShowInputText)
+  const { setValueFromEvent } = useForm()
+  const children = usePropsSelector('children')
+  const test = (event: any) => {
```
Comment (tlenclos) [inline src/components/inspector/controls/ChildrenControl.tsx]:
```suggestion
  const onKeyUp = (event: any) => {
```

File: src/components/inspector/controls/ChildrenControl.tsx [hunk 3]
```diff
@@ -1,26 +1,45 @@
...
 
   return (
     <FormControl htmlFor="children" label="Text">
       <Input
         id="children"
         name="children"
-        autoFocus
         size="sm"
-        value={children || ""}
+        value={children || ''}
```
Comment (shinework) [inline src/components/inspector/controls/ChildrenControl.tsx]: the `usePropsSelector` already returns an empty value as default :)

```suggestion
        value={children}
```

File: src/core/models/app.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ type Overlay = undefined | { rect: DOMRect; id: string; type: ComponentType }
 export type AppState = {
   showLayout: boolean
   showCode: boolean
+  focusInputText: boolean
+  useComponentFocused: boolean
```
Comment (shinework) [inline src/core/models/app.ts]: To remove

File: src/core/models/app.ts [hunk 5]
```diff
@@ -36,6 +40,18 @@ const app = createModel({
         showCode: !state.showCode,
       }
     },
+    toggleInputText(state: AppState): AppState {
+      return {
+        ...state,
+        focusInputText: !state.focusInputText,
+      }
+    },
+    toggleComponentFocused(state: AppState): AppState {
```
Comment (shinework) [inline src/core/models/app.ts]: To remove :)

File: src/core/models/app.ts [hunk 6]
```diff
@@ -5,6 +5,8 @@ type Overlay = undefined | { rect: DOMRect; id: string; type: ComponentType }
 export type AppState = {
   showLayout: boolean
   showCode: boolean
+  focusInputText: boolean
```
Comment (shinework) [inline src/core/models/app.ts]:
```suggestion
  inputTextFocused: boolean
```

File: src/core/selectors/app.ts [hunk 7]
```diff
@@ -3,3 +3,9 @@ import { RootState } from '../store'
 export const getShowLayout = (state: RootState) => state.app.showLayout
 
 export const getShowCode = (state: RootState) => state.app.showCode
+
+export const getFocusedComponent = (id: IComponent['id']) => (
+  state: RootState,
+) => state.app.useComponentFocused && state.components.present.selectedId === id
```
Comment (shinework) [inline src/core/selectors/app.ts]:
```suggestion
) => state.app.inputTextFocused && state.components.present.selectedId === id
```

File: src/core/selectors/app.ts [hunk 8]
```diff
@@ -3,3 +3,9 @@ import { RootState } from '../store'
...
+
+export const getShowInputText = (state: RootState) => state.app.focusInputText
```
Comment (shinework) [inline src/core/selectors/app.ts]:
```suggestion
export const getInputTextFocused = (state: RootState) => state.app.inputTextFocused
```

File: src/hooks/useInteractive.ts [hunk 9]
```diff
@@ -32,6 +36,18 @@ export const useInteractive = (
       event.preventDefault()
       event.stopPropagation()
       dispatch.components.select(component.id)
+      if (focusInput) {
+        dispatch.app.toggleInputText()
+        dispatch.app.toggleComponentFocused(false)
```
Comment (shinework) [inline src/hooks/useInteractive.ts]: to delete

File: src/hooks/useInteractive.ts [hunk 10]
```diff
@@ -32,6 +36,18 @@ export const useInteractive = (
...
+      }
+    },
+    onDoubleClick: (event: MouseEvent) => {
+      event.preventDefault()
+      event.stopPropagation()
+      if (focusInput === false) {
+        dispatch.app.toggleInputText()
+        dispatch.app.toggleComponentFocused()
```
Comment (shinework) [inline src/hooks/useInteractive.ts]: to delete

File: src/hooks/useInteractive.ts [hunk 11]
```diff
@@ -50,5 +66,19 @@ export const useInteractive = (
     }
   }
 
+  if (component.id === selectedId && focusInput) {
```
Comment (shinework) [inline src/hooks/useInteractive.ts]: is it mandatory?

```js
if (hover || selectedId === component.id) {
    props = {
      ...props,
      boxShadow: `${focusInput ? '#4FD1C5' : '#ffc4c7'} 0px 0px 0px 2px inset`,
    }
  }
```

also you can use a softer red like #ffc4c7

File: src/hooks/useInteractive.ts [hunk 12]
```diff
@@ -6,7 +6,8 @@ import {
   getIsSelectedComponent,
   getIsHovered,
 } from '../core/selectors/components'
-import { getShowLayout } from '../core/selectors/app'
+import { getShowLayout, getFocusedComponent } from '../core/selectors/app'
+import { getInputTextFocused } from '../core/selectors/app'
```
Comment (shinework) [inline src/hooks/useInteractive.ts]:
```suggestion
```

File: src/hooks/useInteractive.ts [hunk 13]
```diff
@@ -16,12 +17,14 @@ export const useInteractive = (
   const showLayout = useSelector(getShowLayout)
   const isComponentSelected = useSelector(getIsSelectedComponent(component.id))
   const isHovered = useSelector(getIsHovered(component.id))
+  const focusInput = useSelector(getInputTextFocused)
```
Comment (shinework) [inline src/hooks/useInteractive.ts]:
```suggestion
  const focusInput = useSelector(getFocusedComponent(component.id))

```

File: src/hooks/useInteractive.ts [hunk 14]
```diff
@@ -16,12 +17,14 @@ export const useInteractive = (
...
+  const useComponentFocused = useSelector(getFocusedComponent(component.id))
```
Comment (shinework) [inline src/hooks/useInteractive.ts]:
```suggestion
```

File: src/hooks/useInteractive.ts [hunk 15]
```diff
@@ -30,11 +33,24 @@ export const useInteractive = (
     },
     onMouseOut: () => {
       dispatch.components.unhover()
+      if (focusInput) {
```
Comment (shinework) [inline src/hooks/useInteractive.ts]: it's not necessary to unfocus when the mouse leave th component

**Pull request merged**